include/revng/TupleTree/TupleTreeDiff.h

```diff
@@ -299,10 +304,9 @@ struct llvm::yaml::MappingTraits<T> {
       IO.mapRequired("Path", SerializedPath);
       auto MaybePath = stringAsPath<Model>(SerializedPath);
       if (!MaybePath.has_value()) {
-        ::ErrorList *EL = static_cast<::ErrorList *>(IO.getContext());
+        auto *EL = static_cast<revng::DiffError *>(IO.getContext());
         std::string ErrorMessage = "Path " + SerializedPath + " is invalid";
-        EL->push_back(llvm::createStringError(llvm::inconvertibleErrorCode(),
-                                              ErrorMessage));
+        EL->addReason(ErrorMessage, revng::DiffError::LocationType(*MaybePath));
```

Comment:
You can't inspect `MaybePath` in an `if` where you verified that it does not have a value.